newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js

```diff
@@ -11,54 +11,43 @@ import {
 import type { $AxiosError } from 'axios';
 import type { MessageDescriptor } from '../../Utils/i18n/MessageDescriptor.flow';
 import { serializeToJSON } from '../../Utils/Serializer';
-import { initializeZipJs } from '../../Utils/Zip.js';
 import CloudSaveAsDialog from './CloudSaveAsDialog';
 import { t } from '@lingui/macro';
+import { createZipWithSingleTextFile } from '../../Utils/Zip.js/Utils';
 
 const zipProject = async (project: gdProject) => {
-  const zipJs: ZipJs = await initializeZipJs();
   const projectJson = serializeToJSON(project);
-  const textReader = new zipJs.TextReader(projectJson);
-
-  return new Promise((resolve, reject) => {
-    zipJs.createWriter(
-      new zipJs.BlobWriter('application/zip'),
-      zipWriter => {
-        zipWriter.add('game.json', textReader, () => {
-          zipWriter.close(blob => {
-            resolve(blob);
-          });
-        });
-      },
-      error => {
-        console.error('An error occurred when zipping project', error);
-        reject(error);
-      }
-    );
-  });
+  return createZipWithSingleTextFile(projectJson, 'game.json');
 };
 
 const zipProjectAndCommitVersion = async ({
   authenticatedUser,
   project,
   cloudProjectId,
-}: {
+  options,
+}: {|
   authenticatedUser: AuthenticatedUser,
   project: gdProject,
   cloudProjectId: string,
-}): Promise<?string> => {
+  options?: {| previousVersion: string |},
+|}): Promise<?string> => {
   const archive = await zipProject(project);
-  const newVersion = await commitVersion(
+  const newVersion = await commitVersion({
     authenticatedUser,
     cloudProjectId,
-    archive
-  );
+    zippedProject: archive,
+    previousVersion: options ? options.previousVersion : null,
```

Comment:
does this mean we can only retrieve previous versions from new commits and onwards?

Reply:
I may add a comment to explain why this is useful to provide

Reply:
No, we already save the previous version for each version but when we do it, we use the project current version
In a case where the current version is corrupt, we want to specify that this new version will have a specific parent version that is not the project current version

Reply:
I'll add a comment on the commit method